lib/rubygems/specification_policy.rb

```diff
@@ -152,7 +152,7 @@ def validate_dependencies # :nodoc:
   if #{dep.name} is semantically versioned, use:
     add_#{dep.type}_dependency '#{dep.name}', '~> #{base.join '.'}', '>= #{dep_version}'
   if #{dep.name} is not semantically versioned, you can bypass this warning with:
-    add_#{dep.type}_dependency '#{dep.name}', '>= #{dep_version}', '< #{upper_bound.join '.'}'
+    add_#{dep.type}_dependency '#{dep.name}', '>= #{dep_version}', '< #{upper_bound.join '.'}.x'
```

Comment:
why `.x` as the upper bound? This will continue to allow `#{upper_bound}.a` to be resolved, I believe

Reply:
Not sure why, but it works, I think I saw it somewhere. If I use `< 5.2.x`, it no longer allows `5.2.0.rc1` whereas `< 5.2` allows it.

Reply:
is it because `x` is alphabetically before `0`? 😂

Reply:
Funny I thought this was a feature and it seems like it's a bug  :rofl:.

If there's no "supported" equivalent of the semver operator using ">", "<=", then this advice is actually misleading...

Reply:
I only now understood @seggidins original comment. I'll do some more experiments to confirm if the actual non numeric digit is actually relevant. It seems better to use `.a` anyways since it's already used in other places.